radix-engine/src/blueprints/epoch_manager/epoch_manager.rs

```diff
@@ -293,15 +398,34 @@ impl EpochManagerBlueprint {
         let next_validator_set: BTreeMap<ComponentAddress, Validator> =
             validators.into_iter().collect();
 
-        let handle = api.actor_lock_field(
+        let validator_set_handle = api.actor_lock_field(
             OBJECT_HANDLE_SELF,
             EpochManagerField::CurrentValidatorSet.into(),
             LockFlags::MUTABLE,
         )?;
-        let mut validator_set: CurrentValidatorSetSubstate = api.field_lock_read_typed(handle)?;
-        validator_set.validator_set = next_validator_set.clone();
-        api.field_lock_write_typed(handle, &validator_set)?;
-        api.field_lock_release(handle)?;
+        api.field_lock_write_typed(
+            validator_set_handle,
+            CurrentValidatorSetSubstate {
+                validator_set: next_validator_set.clone(),
+            },
+        )?;
+        api.field_lock_release(validator_set_handle)?;
+
+        let statistic_handle = api.actor_lock_field(
+            OBJECT_HANDLE_SELF,
+            EpochManagerField::CurrentProposalStatistic.into(),
+            LockFlags::MUTABLE,
+        )?;
+        let mut statistic: CurrentProposalStatisticSubstate =
+            api.field_lock_read_typed(statistic_handle)?;
```

Comment:
_(a note from the author)_

Here I definitely want to read and write (see the TODO below, to be addressed soon).